app/controllers/hierarchical_responses_controller.rb

```diff
@@ -0,0 +1,350 @@
+      format.csv do
+        authorize!(:export, Response)
+        @responses = @responses.accessible_by(current_ability, :export)
+
+        # do search, excluding excerpts
+        if params[:search].present?
+          begin
+            @responses = Response.do_search(@responses, params[:search], {mission: current_mission},
+              include_excerpts: false)
+          rescue Search::ParseError
+            flash.now[:error] = $!.to_s
+            return
+          end
+        end
+
+        # Get the response, for export, but not paginated.
+        # We deliberately don't eager load as that is handled in the Results::Csv::Generator class.
+        @responses = @responses.order(:created_at)
+
+        @csv = Results::Csv::Generator.new(@responses)
+        render_csv("elmo-#{current_mission.compact_name}-responses-#{Time.zone.now.to_s(:filename_datetime)}")
+      end
+    end
+  end
+
+  def show
+    # if there is a search param, we try to load the response via the do_search mechanism
+    # so that we get highlighted excerpts
+    if params[:search]
+      # we pass a relation matching only one respoonse, so there should be at most one match
+      matches = Response.do_search(Response.where(id: @response.id), params[:search],
+        {mission: current_mission}, include_excerpts: true, dont_truncate_excerpts: true)
+
+      # if we get a match, then we use that object instead, since it contains excerpts
+      @response = matches.first if matches.first
+    end
+
+    prepare_and_render_form
+  end
+
+  def new
+    setup_condition_computer
+    Results::BlankResponseTreeBuilder.new(@response).build
+    # render the form template
+    prepare_and_render_form
+  end
+
+  def edit
+    flash.now[:notice] = "#{t('response.checked_out')} #{@response.checked_out_by_name}" if @response.checked_out_by_others?(current_user)
+    prepare_and_render_form
+  end
+
+  def create
+    # if this is a non-web submission
+    if request.format == Mime::XML
+      begin
+        @response.user_id = current_user.id
+        # If it looks like a J2ME submission, process accordingly
+        if params[:data] && params[:data][:'xmlns:jrm'] == "http://dev.commcarehq.org/jr/xforms"
+          @submission = XMLSubmission.new response: @response, data: params[:data], source: "j2me"
+        else # Otherwise treat it like an ODK submission
+          upfile = params[:xml_submission_file]
+
+          # Store file for debugging purposes.
+          UploadSaver.new.save_file(params[:xml_submission_file])
+
+          files = params.select { |k, v| v.is_a? ActionDispatch::Http::UploadedFile }
+          files.each { |k, v| files[k] = v.tempfile }
+
+          unless upfile
+            render_xml_submission_failure("No XML file attached.", 422)
+            return false
+          end
+
+          @response.awaiting_media = true if params["*isIncomplete*"] == "yes"
+          @submission = XMLSubmission.new response: @response, files: files, source: "odk"
```

Comment:
Was this branch rebased on develop after 6441 was merged? This method has changed a lot.

Reply:
I don't think so.  This controller is a duplicate of the `ResponsesController` but with a few things changed to keep it working with a `Response` model (as opposed to a `HierarchicalResponse` model).  Perhaps I'll rebase against `develop` and copy this action over again?

Reply:
@scttnlsn That sounds like a good idea. Has it changed at all?

Reply:
Might be a good idea to re-copy any other files that changed in Devney's issue also...

Reply:
Did the rebase and then recreated my changes to a duplicated responses controller.  What other files are you referring to @smoyth?